cpp/src/copying/copy.cu

```diff
@@ -401,7 +408,8 @@ std::unique_ptr<column> copy_if_else(scalar const& lhs,
                                      rmm::cuda_stream_view stream,
                                      rmm::mr::device_memory_resource* mr)
 {
-  CUDF_EXPECTS(lhs.type() == rhs.type(), "Both inputs must be of the same type");
+  CUDF_EXPECTS(
+    lhs.type() == rhs.type(), "Both inputs must be of the same type", cudf::data_type_error);
```

Comment:
This is going to conflict with my PR #14531. I've been trying to revive and finish that one recently but still need to iron out some bits for column-scalar nested type comparisons. There will be some conflicts to resolve, but nothing too difficult.

Reply:
I guess we're racing then :stuck_out_tongue: